NGS generates FASTQ files. FASTQC is an computational approach to evaluate the quality of your NGS data.

FASTQ data format(for each reads):

```
@SRR001666.1 071112_SLXA-EAS1_s_7:5:1:817:345 length=36			# sequence ID
GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACC												# sequence
+SRR001666.1 071112_SLXA-EAS1_s_7:5:1:817:345 length=36			# quality ID
IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9IC												# sequence score
```

The byte representing quality runs from 0x21 (lowest quality; '!' in ASCII) to 0x7e (highest quality; '~' in ASCII). Here are the quality value characters in left-to-right increasing order of quality ([ASCII](https://en.wikipedia.org/wiki/ASCII)):

```
 !"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~
```

Why do we need to do quality control? FASTQ files are massive.

How to interpret FASTQC results?

1. A quality value *Q* should be above 25 

2. Per sequence quality distribution should be uniform, and average quality score should be above 30

3. Nucleotide content per position should be smooth over the length
4. Per sequence GC content should follow normal distribution

A quality value *Q* is an integer mapping of *p* (i.e., the probability that the corresponding base call is incorrect). Two different equations have been in use. The first is the standard Sanger variant to assess reliability of a base call, otherwise known as [Phred quality score](https://en.wikipedia.org/wiki/Phred_quality_score):

Qsanger=−10log10⁡p![Q_{\text{sanger}}=-10\,\log _{10}p](https://wikimedia.org/api/rest_v1/media/math/render/svg/d35e9473c8fb52974f7f37c6bd66852e2e276da3)

Normally the quality score decreases with the length, but not too much



Reference:

[FASTQ format](https://en.wikipedia.org/wiki/FASTQ_format)

[FASTQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)