1. Early sequence alignment (1 with 1)

Global alignment: the Needlemen-Wunsch algorithm is the earliest algorithm to find the alignment between two sequences and score their similarity.

Local alignment: when two sequences are long, and only a portion of them can align well with each other, the Smith-Waterman algorithm can find the best local sequence alignment. It is still considered the best alignment approach, although it is slow.

2. Sequence search algorihtms (1 with many)

The fast search algorithm BLAST was developed in 1980s, using one sequence as the query to find similar sequences from a database.

BLAST: seed(k-mer)

STAR: suffix array: [后缀数组](https://so.csdn.net/so/search?q=后缀数组)，一种处理字符串的有力工具，用于提高查找比对的效率。

3. Borrow-Wheeler Aligner (many with many)

With NGS, scientists need much faster search (aka mapping) algorithms in order to align the millions of sequences to the reference genome. The current best algorithm is called Borrow-Wheeler Aligner or BWA.

Borrow-Wheeler transformation and LF mapping

Borrow-Wheeler aligners: bwa and bowtie



Reference

[Global alignment](https://youtu.be/ipp-pNRIp4g)

[Local alignment](https://youtu.be/lu9ScxSejSE)

[后缀数组(suffix array)详解](https://blog.csdn.net/u013371163/article/details/60469533)

[后缀数组(Suffix Array)在字符串匹配中的应](https://segmentfault.com/a/1190000021460167)

