NGS raw data is in FASTQ. Alignment gives you SAM (alignment) or BAM (binary version of SAM) files which contain the sequence information in FASTQ and the mapping locations. BED file is the simpliest, although there is information loss.

The **SAM Format** is a text format for storing sequence data in a series of tab delimited ASCII columns.

![Screenshot 2022-01-10 at 17.43.17](/Users/u1995087/Desktop/Screenshot 2022-01-10 at 17.43.17.png)

Mapped seq files in SAM file

![Screenshot 2022-01-10 at 17.44.00](/Users/u1995087/Desktop/Screenshot 2022-01-10 at 17.44.00.png)

A **BAM file **(. bam) is the binary version of a SAM file. 

Visualization in a genome browser

![Screenshot 2022-01-10 at 17.45.20](/Users/u1995087/Desktop/Screenshot 2022-01-10 at 17.45.20.png)

The **BED** (**Browser Extensible Data**) format is a [text file](https://en.wikipedia.org/wiki/Text_file) format used to store [genomic](https://en.wikipedia.org/wiki/Genome) regions as [coordinates](https://en.wikipedia.org/wiki/Coordinate_system) and associated [annotations](https://en.wikipedia.org/wiki/DNA_annotation). 

![Screenshot 2022-01-10 at 17.45.48](/Users/u1995087/Desktop/Screenshot 2022-01-10 at 17.45.48.png)

The **bigBed** format stores annotation items that can be either a simple or a linked collection of exons, much as [BED](https://genome.ucsc.edu/FAQ/FAQformat.html#format1) files do. BigBed files are created from BED type files using the program `bedToBigBed`. The resulting bigBed files are in an indexed binary format. 



Reference

[SAM](https://samtools.github.io/hts-specs/SAMv1.pdf)

[BAM](https://support.illumina.com/help/BS_App_RNASeq_Alignment_OLH_1000000006112/Content/Source/Informatics/BAM-Format.htm)

[BED](https://genome.ucsc.edu/FAQ/FAQformat.html#format1)

[bigBED](https://genome.ucsc.edu/goldenPath/help/bigBed.html)